graphql/src/main/java/com/linecorp/armeria/server/graphql/DefaultGraphQLService.java

```
@@ -0,0 +1,194 @@
+                }
+                final Map<String, Object> requestMap = parseJsonString(body);
+                final String query = (String) requestMap.getOrDefault("query", "");
+                return execute(executionInput(query, ctx,
+                                              toVariableMap(requestMap.get("variables")),
+                                              (String) requestMap.get("operationName")));
+            }));
+        }
+        if (contentType.is(MediaType.GRAPHQL)) {
+            return HttpResponse.from(request.aggregate().handle((req, thrown) -> {
+                if (thrown != null) {
+                    return HttpResponse.ofFailure(thrown);
+                }
+                final String query = req.contentUtf8();
+                return execute(executionInput(query, ctx, null, null));
+            }));
+        }
+        return HttpResponse.of(HttpStatus.UNPROCESSABLE_ENTITY,
+                               MediaType.PLAIN_TEXT,
+                               "Could not process GraphQL request");
+    }
+
+    private ExecutionInput executionInput(String query, ServiceRequestContext ctx,
+                                          @Nullable Map<String, Object> variables,
+                                          @Nullable String operationName) {
+        final Builder builder = ExecutionInput.newExecutionInput(query);
+        if (variables != null && !variables.isEmpty()) {
+            builder.variables(variables);
+        }
+        if (operationName != null) {
+            builder.operationName(operationName);
+        }
+        if (dataLoaderRegistry != null) {
+            builder.dataLoaderRegistry(dataLoaderRegistry);
+        }
+        return builder.context(ctx).build();
+    }
+
+    private HttpResponse execute(ExecutionInput input) {
+        return HttpResponse.from(graphQL.executeAsync(input).handle((executionResult, cause) -> {
```

We can not support Graphql Subscription at the moment until WebSockets is implemented.
- Can we throw `UnsupportedOperationException` when using Graphql Subscription?
- Add TODO for Graphql Subscription?